client/deposits/declarations.d.ts

```diff
@@ -29,4 +29,11 @@ declare module '@woocommerce/components' {
 		orderStatusMap: Object;
 		labelPositionToLeft?: boolean;
 	} ) => JSX.Element;
+
```

Comment:
Type defs for Link were only added to `@woocommerce/components` recently: https://github.com/woocommerce/woocommerce/pull/36285.

Since we are using a [legacy version of this library (5.1.2)](https://github.com/Automattic/woocommerce-payments/blob/322346fd584abd7b62ae04fa0b1670c3798cb859/package.json#L96), it will require updating before we can access these type defs.

Note: I didn't add this as a new declaration, I moved it from `client/deposits-information/declarations.d.ts` to a shared location: `client/deposits/declarations.d.ts`.

Reply:
When we have to supply definitions for a library, I think we should explicitly note that in a comment.

e.g. 

- This type is from library @bob/cool-components which doesn't have type definitions (in v1.2.3 that we use).
- Types are coming in v4.2 and when we update we should remove this. 

i.e. something to capture the insights from the discussion in PR right in the code.

Reply:
(this is great FE guild material!)

Reply:
After investigating further, it doesn't look like types for the `Link` component have been released yet: https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/changelog/dev-migrate-link-ts.

> I think we should explicitly note that in a comment.
I've added a comment to this effect in 0d9631bdd9bc233429ce4b26ecf8faf079b6edfc, thanks @haszari.

Reply:
Random idea (maybe for later): define lib types in a separate module, as a clear namespace & association with upstream/third party. I.e. delete the file when updating to lib with types.

Reply:
Agreed! At the moment this module is declared in both `client/deposits` and [`client/transactions`](https://github.com/Automattic/woocommerce-payments/blob/0d9631bdd9bc233429ce4b26ecf8faf079b6edfc/client/transactions/declarations.d.ts#L32), but the components declared are used throughout the codebase.